tests/e2e/network.go

```diff
@@ -0,0 +1,128 @@
+// Copyright (C) 2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package e2e
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/tests/fixture/e2e"
+
+	"github.com/ava-labs/hypersdk/api/indexer"
+	"github.com/ava-labs/hypersdk/api/jsonrpc"
+	"github.com/ava-labs/hypersdk/chain"
+	"github.com/ava-labs/hypersdk/tests/workload"
+)
+
+var (
+	ErrUnableToConfirmTx = errors.New("unable to confirm transaction")
+	ErrInvalidURI        = errors.New("invalid uri")
+)
+
+const (
+	txCheckInterval = 100 * time.Millisecond
+)
+
+type Network struct {
+	nodes  []*Node
+	parser *parser
+}
+
+func NewNetwork(tc *e2e.GinkgoTestContext) *Network {
+	blockchainID := e2e.GetEnv(tc).GetNetwork().GetSubnet(networkConfig.Name()).Chains[0].ChainID
+	testNetwork := &Network{}
+	for _, uri := range getE2EURIs(tc, blockchainID) {
+		n := &Node{uri: uri, network: testNetwork}
+		testNetwork.nodes = append(testNetwork.nodes, n)
+	}
+	testNetwork.parser = &parser{
+		Parser: networkConfig.Parser(),
+		rules: &rules{
+			Rules:   networkConfig.Parser().Rules(0),
+			chainID: blockchainID,
+		},
+	}
+	return testNetwork
+}
+
+func (n *Network) URIs() (out []string) {
+	for _, node := range n.nodes {
+		out = append(out, node.URI())
+	}
+	return
+}
+
+func (n *Network) ConfirmTxs(ctx context.Context, txs []*chain.Transaction) error {
+	return n.nodes[0].ConfirmTxs(ctx, txs)
+}
```

Comment:
This function should make sure that txs are confirmed on all of the nodes in the network. Could we update this to:
1. Issue txs on 1 node
2. Confirm txs have been accepted on that one node
3. Iterate over the remaining nodes and confirm that it has advanced to the same block height/blockID (which should guarantee that it's accepted the same transactions)

Alternatively to 3, we could manually confirm each tx via API call on each node.

Reply:
done